service/common/src/main/java/org/apache/polaris/service/catalog/io/DefaultFileIOFactory.java

```diff
@@ -70,13 +71,14 @@ public DefaultFileIOFactory(
 
   @Override
   public FileIO loadFileIO(
-      @Nonnull RealmContext realmContext,
+      @Nonnull CallContext callContext,
```

Comment:
`CallContext` feels like a request-scoped bean... Could we have it injected as opposed to passed as a method parameter?

Reply:
Because of the existence of "system-initiated" actions that will need to become better formalized in the future, I'm trying to minimize the use of `@RequestScoped` for the CallContext object itself in favor of explicitly passing it through for now, at least until we sort out how to portray the relationship between API-initiated requests, "async actions" that are still tied to an original request, and fully system-initiated background actions.